### DISCUSSION TITLE
Guidance for Using ONNX with Custom C++ Models

### DISCUSSION THREAD

**Discussion by obewan** [microsoft/onnxruntime#18147] [Other Q&A]
Hi,

I have a custom neural network implemented in C++ and I’m interested in exporting this network to ONNX format. I’m currently using the ONNX Runtime C++ API, but I’m finding it challenging to convert my custom network into an ONNX model.

I would prefer to stick with C++ for this task and not use Python. Could you provide some guidance or point me to any resources that might help with this process, specifically within a C++ context?

Best regards,
Damien

**Comment by Craigacp** [comment]
The ONNX format is a protobuf, so you can write it directly using the C++ Protobuf library from your custom implementation. The spec is documented here - https://github.com/onnx/onnx/blob/main/docs/IR.md.

**Reply by obewan** [reply]
Thanks for the infos, it will help.
